### PR TITLE
Adds generator for complete multipartite graph.

### PR DESCRIPTION
**Pull request by jfinkels**
This is a generalization of the complete bipartite graph generator.

### CODE REVIEW

**Comment by chebee7i** [comment]
Can you add a note in the api_2.0 release notes?

**Comment by jfinkels** [comment]
Added note to `api_2.0.rst`.

File: networkx/generators/classic.py [hunk 1]
```diff
@@ -490,3 +511,37 @@ def wheel_graph(n,create_using=None):
         G.add_edge(1,n-1)
     return G
 
+
+# TODO On Python 3, we can define this as
+#
+#     def complete_multipartite_graph(*block_sizes, create_using=None):
+#
+def complete_multipartite_graph(*block_sizes, **kw):
+    """Returns the complete multipartite graph with the specified block sizes.
```
Comment (chebee7i) [inline networkx/generators/classic.py]: Sorry, one last thing. Can you clean up this docstring to make it more in line with how we typically have them in NetworkX. So a Parameters, Returns, and Examples section---the examples section will be helpful for people see esp when dealing with the `*block_sizes`.

Comment (jfinkels) [inline networkx/generators/classic.py]: Oh sure, I can get to this tomorrow. This is just how I write my own docstrings; force of habit.

**Comment by jfinkels** [comment]
Updated docstring. Also, added a `block` attribute to each node, similar to the `bipartite` attribute from the `complete_bipartite_graph` generator.

**Comment by jtorrents** [comment]
@jfinkels very nice implementation. I have a comment on the interface. I'm not sure that is a good idea to have a `create_using` parameter (which, by the way, is a pretty ugly interface in general). If an user uses as argument a directed graph, the directions of the edges between blocks will depend on the number of blocks and on the iteration order of `itertools.combinations`. 

Also the comment just above the combinations loop says that we can use combinations because we consider the multipartite graph undirected. I think this is a good approach, it's just a matter to remove the `create_using` parameter and return always an undirected graph. What do you think?

Finally, could you rebase and solve the trivial conflict with `api_2.0.rst`?

**Comment by jfinkels** [comment]
> I'm not sure that is a good idea to have a `create_using` parameter (which, by the way, is a pretty ugly interface in general). If an user uses as argument a directed graph, the directions of the edges between blocks will depend on the number of blocks and on the iteration order of `itertools.combinations`.

The `create_using` idiom is ugly, I agree; I recently objected to it in #1393. I only added it here because the other functions in this module had it. I'm happy to remove it.

**Comment by jtorrents** [comment]
Yes, we discussed removing it altogether some time ago. If I remember correctly the discussion, the alternative in most cases could simply be a boolean parameter `directed` as in `nx.gnp_random_graph`.

The changes look good to me. Let's see if someone else wants to chime in about removing the `create_using` parameter. If not I'll merge it in the following days.

**Comment by hagberg** [comment]
Yes, let's not use create_using.  See #1393 for discussion on reasons.

I like the use of accumulate.  We also have  networkx.utils.cumulative_sum() which will do the same thing without adding more code.  Maybe cumulative_sum() uses should instead be replaced with accumulate?

**Comment by jfinkels** [comment]
@hagberg Well, `accumulate()` is only available on Python 3.2 or later, so as long as you support Python 2.7 you'll have to do the ugly import check wherever you wish to use `accumulate()`. However, I think I prefer that to having a custom `cumulative_sum()` function. After a quick search it only appears in two places, `algorithms.richclub` and `generators.geometric`.

**Comment by hagberg** [comment]
How about putting the accumulate import ugliness in networkx.utils.misc.accumulate and we'll using that function until eventually we deprecate Python 2?

**Comment by jfinkels** [comment]
Fair enough, I can create a different pull request for that.

**Pull request merged**